src/test/java/com/epam/aidial/core/FileApiTest.java

```diff
@@ -0,0 +1,269 @@
+package com.epam.aidial.core;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.epam.aidial.core.config.Storage;
+import com.epam.aidial.core.data.FileMetadata;
+import com.epam.aidial.core.storage.BlobStorage;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.codec.BodyCodec;
+import io.vertx.ext.web.multipart.MultipartForm;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import lombok.extern.slf4j.Slf4j;
+import org.jclouds.filesystem.reference.FilesystemConstants;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.nio.file.Path;
+import java.util.Properties;
+
+@ExtendWith(VertxExtension.class)
+@Slf4j
+public class FileApiTest {
+
+    private static final String TEST_FILE_CONTENT = "Test file content";
+
+    private static AiDial dial;
+    private static int serverPort;
+    private static Path testDir;
+
+    @BeforeAll
+    public static void init() throws Exception {
+        // initialize server
+        dial = new AiDial();
+        testDir = TestFiles.baseTestPath(FileApiTest.class);
+        BlobStorage storage = buildFsBlobStorage(testDir);
+        dial.start(storage);
```

Comment:
I would pass a path to dial config as cmd arg to DIAL app instead of exposing one more method.
See com.epam.aidial.core.AiDial#settings()